pages/docs/concepts/asyncapi-document/variable-url.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Variables in URL
+weight: 235
+---
+
+Event-driven architecture benefits from a defined base URL and rules for URL variables. You can easily manage multiple API endpoints, handling various server configurations and environments.
+
+URL variables are placeholders for values you can replace during runtime. Async API enables you to construct dynamic URLs with query parameters and additional information while enhancing the flexibility and maintainability of your API specifications.
+
+This article assumes that you have a basic understanding of AsyncAPI. For more information, refer to the [Event-Driven Architectures](../../tutorials/getting-started/event-driven-architectures.md).
+
+## Add Dynamic Variables
+
+To add variables to the URL using AsyncAPI between two servers. You can use the `server.url` and `components.serverVariables` fields to enable reusable variables across multiple servers.
+
+The diagram below depicts how to construct dynamic URLs with query parameters.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style E fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+style F fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[AsyncAPI Document]
+B[Servers Section]
+C[Base URLs]
+D[components.serverVariables Section]
+E[Domain Variable]
+F[Port Variable]
+
+A -->|contains| B
+A -->|contains| D
+B -->|includes| C
+D -->|defines| E
+D -->|defines| F
+```
```

Comment:
A few things:
* What's the purpose of this diagram?
* Variables can be used for all sorts of things not just domains and ports.
* Variables can be defined directly under the server without using `components.serverVariables`.

Reply:
In today's Spec 3 Docs meeting, I addressed Fran's question about why we have diagrams in our project.